lbrynet/conf.py

```diff
@@ -137,56 +75,118 @@ def update(self, other):
+            for key, value in schema.items()
+        }
+        envparse.Env.__init__(self, **my_schema)
+
+    def __call__(self, key, *args, **kwargs):
+        my_key = self._convert_key(key)
+        return super(Env, self).__call__(my_key, *args, **kwargs)
+
+    def _convert_key(self, key):
+        return Env.NAMESPACE + key.upper()
+
+    def _convert_value(self, value):
+        """Allow value to be specified as an object, tuple or dict
+
+        if object or dict, follow default envparse rules, if tuple
+        it needs to be of the form (cast, default) or (cast, default, subcast)
+        """
+        if isinstance(value, dict):
+            return value
+        if isinstance(value, (tuple, list)):
+            new_value = {'cast': value[0], 'default': value[1]}
+            if len(value) == 3:
+                new_value['subcast'] = value[2]
+            return new_value
+        return value
+
+
+def server_port(server_port):
+    server, port = server_port.split(':')
+    return server, port
+
+
+DEFAULT_DHT_NODES = [
+    ('lbrynet1.lbry.io', 4444),
+    ('lbrynet2.lbry.io', 4444),
+    ('lbrynet3.lbry.io', 4444)
+]
+
+
+ENVIRONMENT = Env(
+    is_generous_host=(bool, True),
+    run_on_startup=(bool, False),
+    download_directory=(str, default_download_directory),
+    max_upload=(float, 0.0),
+    max_download=(float, 0.0),
+    upload_log=(bool, True),
+    delete_blobs_on_remove=(bool, True),
+    use_upnp=(bool, True),
+    start_lbrycrdd=(bool, True),
+    run_reflector_server=(bool, False),
+    startup_scripts=(list, []),
+    # TODO: this doesn't seem like the kind of thing that should
+    # be configured; move it elsewhere.
+    last_version=(dict, {'lbrynet': '0.0.1', 'lbryum': '0.0.1'}),
+    peer_port=(int, 3333),
+    dht_node_port=(int, 4444),
+    reflector_port=(int, 5566),
+    download_timeout=(int, 30),
+    max_search_results=(int, 25),
+    search_timeout=(float, 3.0),
+    cache_time=(int, 150),
+    host_ui=(bool, True),
+    check_ui_requirements=(bool, True),
+    local_ui_path=(bool, False),
+    api_port=(int, 5279),
+    search_servers=(list, ['lighthouse1.lbry.io:50005']),
+    data_rate=(float, .0001),  # points/megabyte
+    min_info_rate=(float, .02),  # points/1000 infos
+    min_valuable_info_rate=(float, .05),  # points/1000 infos
+    min_valuable_hash_rate=(float, .05),  # points/1000 infos
+    max_connections_per_stream=(int, 5),
+    known_dht_nodes=(list, DEFAULT_DHT_NODES, server_port),
+    pointtrader_server=(str, 'http://127.0.0.1:2424'),
+    reflector_servers=(list, [("reflector.lbry.io", 5566)], server_port),
+    wallet=(str, LBRYUM_WALLET),
+    ui_branch=(str, "master"),
+    default_ui_branch=(str, 'master'),
+    data_dir=(str, default_data_dir),
+    lbryum_wallet_dir=(str, default_lbryum_dir),
+    use_auth_http=(bool, False),
+    sd_download_timeout=(int, 3),
+    # TODO: document what the 'address' field is used for
```

Comment:
[It's because the max key fee gets converted into LBC by the exchange rate manager, and beforehand it gets passed through the Fee validator. ](https://github.com/lbryio/lbry/blob/76fb7697fff8dedc853312276d72f8c92ce33562/lbrynet/lbrynet_daemon/ExchangeRateManager.py#L161-L161) Perhaps the thing to do is refactor the exchange rate conversion in a later pull request to take a currency and an amount rather than a "real" fee dict.